src/main.rs

```diff
@@ -138,9 +155,19 @@ async fn producers(brokers: String, topic: String, m: usize, n: usize, propertie
         )))
     }
 
+    let mut results = vec![];
     for t in tasks {
-        t.await.unwrap();
+        results.push(t.await.unwrap());
     }
+
+    let min_result = *results.iter().min().unwrap();
```

Comment:
nit: this will give an ugly panic if someone ran with 0 produces (which doesn't make sense, but isn't explicitly prevented in argument parsing.

Reply:
Good point, updated the PR to enure results isn't empty.